### DISCUSSION TITLE
Ability to create "many-to-one-to-many" relationship based on shared key that is not primary key in another table

### DISCUSSION THREAD

**Discussion by PMLP-novo** [sqlalchemy/sqlalchemy#9109] [Usage Questions]
### Describe the use case

I have an application that collects data from different sources, which I then combine. The problem is that I run into a situation where the shared key `outside_id` is not the primary key of any existing table that I have access to. What I want is to be able to create the relationship between the two tables and be able to query it eagerly without creating and maintaining a mapping table  solely for the purpose of this relationship. 

### Databases / Backends / Drivers targeted

I'm using postgres, but it could be relevant for all I think. 

### Example Use

```
class B(model):
    id = Column(Integer(), primary_key=True, nullable=False)
    outside_id = Column(String())

class A(model):
    id = Column(Integer(), primary_key=True, nullable=False)
    outside_id = Column(String())

    b_list = relationship(
        "B",
        foreign_keys=[outside_id],
        primaryjoin="A.outside_id==B.outside_id",
        viewonly=True,
    )
```

**Comment by zzzeek** [comment]
im having trouble parsing what you mean, but I guess you mean, "outside_id is not a primary key column".  that's fine, relationship can work with any set of columns as the criteria, you just use the foreign() annotation on the foreign column

```py

from sqlalchemy.orm import foreign

class A(model):
    id = Column(Integer(), primary_key=True, nullable=False)
    outside_id = Column(String())

    b_list = relationship(
        "B",
        foreign_keys=[outside_id],
        primaryjoin="A.outside_id==foreign(B.outside_id)",
        viewonly=True,
    )

```
